crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -862,25 +877,51 @@ unsafe impl<'a, T: Resource> SystemParam for Res<'a, T> {
         world: UnsafeWorldCell<'w>,
         change_tick: Tick,
     ) -> Self::Item<'w, 's> {
-        let (ptr, ticks, caller) =
-            world
-                .get_resource_with_ticks(component_id)
-                .unwrap_or_else(|| {
-                    panic!(
-                        "Resource requested by {} does not exist: {}",
-                        system_meta.name,
-                        core::any::type_name::<T>()
-                    )
-                });
-        Res {
-            value: ptr.deref(),
-            ticks: Ticks {
+        unsafe fn get_param<'w>(
```

Comment:
Not really related to this PR, but I was surprised to see that you updated `Res::get_param` but not `ResMut::get_param`.  It looks like that's because the common code there was already extracted to a `get_resource_mut_by_id` method.  I agree this is the right thing for now, but I wonder if we'll want to add a `struct RefUntyped` someday so that we can have a similar method for `Ref`.